core/nuxeo-document.js

```diff
@@ -226,32 +226,59 @@ import './nuxeo-resource.js';
       return path;
     }
 
-    appendClientReason(prop) {
-      prop.viewData = `${prop.data}&clientReason=view`;
-      prop.downloadData = `${prop.data}&clientReason=download`;
+    appendClientReason(prop, subProp, param) {
+      prop[subProp === 'url' ? 'viewUrl' : 'viewData'] = `${prop[subProp]}${param.view}`;
+      prop[subProp === 'url' ? 'downloadUrl' : 'downloadData'] = `${prop[subProp]}${param.download}`;
     }
 
     setDocumentViewDownloadProp() {
       const documentProps = [];
-      if (this.documentData && this.documentData.properties) {
-        documentProps.push(
-          this.documentData.properties['file:content'],
-          this.documentData.properties['files:files'],
-          this.documentData.properties['picture:views'],
-          this.documentData.properties['vid:transcodedVideos'],
-          this.documentData.properties['vid:storyboard'],
-        );
-        documentProps.forEach((docProp) => {
-          if (docProp) {
-            if (Array.isArray(docProp)) {
-              docProp.forEach((prop) => {
-                this.appendClientReason(prop.file ? prop.file : prop.content);
-              });
-            } else {
-              this.appendClientReason(docProp);
-            }
+      if (this.documentData) {
+        if (this.documentData.contextParameters) {
+          const documentContextParameters = this.documentData.contextParameters;
+          if (documentContextParameters.preview && documentContextParameters.preview.url) {
+            const documentPreview = documentContextParameters.preview;
+            const param = {
+              view: '&clientReason=view',
+              download: '&clientReason=download',
+            };
+            this.appendClientReason(documentPreview, 'url', param);
+          }
+          if (documentContextParameters.renditions) {
+            documentContextParameters.renditions.forEach((rendition) => {
+              const param = {
+                view: '?clientReason=view',
+                download: '?clientReason=download',
+              };
+              this.appendClientReason(rendition, 'url', param);
+            });
           }
-        });
+        }
+        if (this.documentData.properties) {
+          const documentDataProperties = this.documentData.properties;
+          documentProps.push(
+            documentDataProperties['file:content'],
+            documentDataProperties['files:files'],
+            documentDataProperties['picture:views'],
+            documentDataProperties['vid:transcodedVideos'],
+            documentDataProperties['vid:storyboard'],
+          );
+          documentProps.forEach((docProp) => {
+            if (docProp) {
+              const param = {
+                view: '&clientReason=view',
+                download: '&clientReason=download',
+              };
```

Comment:
Kindly remove repetitive creation of param variable. It should be inside appendClientReason function